### PR TITLE
[wip] refactor: articles pagination

### PR DESCRIPTION
**Pull request by alfonsobries**
## Summary

Closes https://app.clickup.com/t/85zu43hpu

## Checklist

-   [ ] I checked my UI changes against the design and there are no notable differences
-   [ ] I checked my UI changes for any responsiveness issues
-   [ ] I checked my (code) changes for obvious issues, debug statements and commented code
-   [ ] I opened a corresponding card on Clickup for any remaining TODOs in my code
-   [ ] I added a short description on how to test this PR _(if necessary)_
-   [ ] I added a storybook entry for the component that was added _(if necessary)_
-   [ ] Documentation _(if necessary)_
-   [ ] Tests _(if necessary)_
-   [ ] Ready to be merged

### CODE REVIEW

**Comment by shahin-hq** [comment]
@alfonsobries 
I checked the changes but it doesn't make sense to add this much complexity only for loading data in client side, the `show` method is bloated with too much logic, page limits of articles, activities, NFTs, their filters etc. Instead we could touch the `Pagination` component and add the `setPage` method there, when page changes it'd trigger the filters in the `ArticlesTab` component and load the data, that is it. Otherwise this may cause pollution in other controllers when we need to load data in client-side; while with the `setPage` approach we'd have more scoped (single responsibility) controllers, components. 

I am not mentioning the perks of the `use-query` like caching and control over the data. I'll prepare a small demo and then we can discuss. 

Also related: https://github.com/ArdentHQ/dashbrd/pull/205

File: resources/js/Pages/Collections/Components/Articles/ArticlePagination.tsx [hunk 1]
```diff
@@ -11,7 +11,7 @@ interface Properties {
 export const ArticlePagination = ({ pagination, onPageLimitChange }: Properties): JSX.Element => {
     const { t } = useTranslation();
 
-    if (pagination.meta.total < 12) {
+    if (pagination.meta.total < pagination.meta.per_page) {
```
Comment (shahin-hq) [inline resources/js/Pages/Collections/Components/Articles/ArticlePagination.tsx:14]: > We only show this if there is more than 12 articles associated with collection. If less or equal we hide it. By default we have 12 selected on for how many items we show per page. Copy over logic for how we do for NFTs already. Options are 12, 24, 48, 96.

`pagination.meta.total < pagination.meta.per_page` this check will hide the pagination when `pageLimit=24` and `total = 18`

**Comment by shahin-hq** [comment]
I prepared demo: https://github.com/ArdentHQ/dashbrd/pull/216/files
Since we don't have enough articles I set `pageLimit = 2` but it should be sufficient for testing.
Also skipped `prev` and `next` functionality for now, surely I will implement if this solution is acceptable.

This video show how it work with
- collection articles page
- articles page
- NFTs page - to ensure it is backward compatible
 
https://github.com/ArdentHQ/dashbrd/assets/132887516/bdf4d946-477c-4ce3-8440-90c5ce143a41

**Comment by alfonsobries** [comment]
Closing this pr, later will make a ticket with some ideas on how to refactor collection page but want to wait unitl other tasks are finished

**Pull request closed without merging**